Actions/DetermineProjectsToBuild/DetermineProjectsToBuild.psm1

```diff
@@ -1,68 +1,79 @@
 ﻿<#
-.Synopsis
-    Creates buils dimensions for a list of projects.
-
-.Outputs
-    An array of build dimensions for the projects and their corresponding build modes.
-    Each build dimension is a hashtable with the following keys:
-    - project: The name of the AL-Go project
-    - buildMode: The build mode to use for the project
+    .Synopsis
+        Gets the modified files in a GitHub pull request.
```

Comment:
IMO - these are just individual helperfunctions and doesn't belong in a module,
They are only used from within the action .ps1 script and only to make reading code easier.
Module implies variable scope and if you want to use functions from AL-Go-Helper or GitHub-Helper you need to re-import other modules and scripts.
These modules then live in memory multiple times.
Importing modules and calling functions in modules is also much slower than just using helperfunctions.

Now there can be many reasons why we should structure things in modules - but a list of helperfunctions isn't one IMO

Beside that, using dashes in function names makes PowerShell apply rules to the naming (which then are excluded globally because of this) - why not just use function names like GetModifiedFiles in a helperfunction script?

Reply:
I see we haven't really settled on how to organize helper functions. 
Why is it AL-Go-Helper.**ps1** and Github-Helper.**psm1**?

PS modules provide encapsulation and that's more important than the performance gain (that is not much and it's not like these modules are imported so many times anyway).
Using scripts for helper function creates a mess and it's matter of time to have unwanted function overwrites. Not that this cannot happen with modules, but it's way less likely, especially if there are structured well.

My idea of having a PS modules (not necessarily related to an action) is to provide good interface of functions to be used both in and outside actions. 

In this case `Get-ModifiedFiles` could be a one liner (`gh pr diff $prNumber--repo $repo --name-only`) and possibly better suited for the Github-Helper module, but I didn't want to change it in this PR.

Reply:
I get that modules should be used for stuff that are used in multiple places, but putting 5 functions that are used only by one script in a module doesn't provide any value IMO. The action in itself is already encapsulated.
Moving reusable code to global modules makes a ton of sense. Having a .ps1 and a .psm1 side by side in all actions doesn't IMO.

The action script imports github-helper module and you now have to import github-helper in the function in the module as well - that is the part I don't like.

BTW, I agree that AL-Go-Helper should have been a module, and GitHub-Helper should have been part of this module, bad choices where made, we can change those.

Reply:
In the ReadSecrets, I think there is a reason for the encapsulation (module variables etc)
TBH - I didn't notice the sign module, but I just comment on this because this is a change, which I don't think is necessary and I don't think is the right thing to do.
If you feel strongly about it - then resolve the conversation and discussion is over.

Reply:
> I believe it's a good approach to have some function separation and encapsulation.

I tend to agree with this. Unless using Powershell Modules are significantly slower, I don't see any issues with using them in an action like this one.